4813 DAPI lagre brev som utkast

janitamarielle opened the pull request:
Bruker de nye endepunktene for å lagre og hente utkast. Også endel logikk på å fylle ut felt basert på utkastet. 

Melosys-api: https://github.com/navikt/melosys-api/pull/1648
https://jira.adeo.no/browse/MELOSYS-4813

janitamarielle committed:
Ramme rundt visning av utkast med dummydata

janitamarielle committed:
Design til utkast-boksen

janitamarielle committed:
Fjerner unødvendig useState. Endrer litt types i OpprettBrevReqDto. Setter endel felt basert på dto-objekt

janitamarielle committed:
Endringer i redux-felt oppdaterer nå den inner html-editoren.
Endret key-en til valgAlternativer til å være kode istedenfor term, dette endrer ikke på hva vi sender til backend. Får se om dette får konsekvenser.
Trengte ikke logikk for resetting av type når vi gikk over til useEffects for å sette felt.
Trakk ut logikk for settFeltForFritekstTittel ut i egen funksjon siden det er endel logikk.
OCDen fikk meg til å mekke litt imports i rekkefølge.

janitamarielle committed:
html-bug

janitamarielle committed:
Sett felt for innledningfritekst

janitamarielle committed:
Pitstop commit før etater skrives om til redux.

janitamarielle committed:
Sette etater fra formvalues

janitamarielle committed:
Lager midlertidig logikk for om det er arbeidsgiver/virksomhet eller annen organisasjon

janitamarielle committed:
Tar ibruk faktiske backend-endepunkt

janitamarielle committed:
Flytter utkast-kall i egen fil

janitamarielle committed:
Trekker all logikk inn i hentBrevRequest og bruker enum

janitamarielle commented:
Skal legge inn toggle: https://unleash.nais.io/#/features/strategies/melosys.utkast

janitamarielle committed:
Legger til toggling ved henting og lagring av utkast. Sletting og oppdatering er allerede gjemt av sjekk på aktivtUtkast?.utkastID

janitamarielle committed:
Legger til oppførsel som skjer etter lagre/sletting/opprettelse av utkast. Endret buggy feltlagring for dokumenttittel. Legger til behandlingID i kall til backend der det manglet

martonSkjevelandNav commented:
Boksen med lagrede utkast som vises i `Send Brev` fanen skal vel også vises i `Dokumenter` fanen (ref skissene). Det gjør den ikke nå som jeg kan se

martonSkjevelandNav commented:
Får en error log når jeg trykker `Lagre og fortsett senere`: `SyntaxError: Unexpected end of JSON input`, men ser ikke at det faktisk skjer noe feil

janitamarielle commented on src/felleskomponenter/sideDialog/sendBrev/brevutkast.tsx:
```diff
@@ -0,0 +1,143 @@
+  const tittelTilUtkast = (utkast: Api.Brevutkast.BrevutkastResDto) =>
+    !Utils._isEmpty(utkast.brevbestilling.dokumentTittel)
+      ? utkast.brevbestilling.dokumentTittel
+      : KV.finnTermFraListe(MKV.KTObjects.brev.produserbaredokumenter, utkast.brevbestilling.produserbardokument?.kode);
```
Her har jeg blingsa når jeg merket produserbardokument var KTObject. Kan jo bare ta termen fra objektet direkte 🤦 
Undersøker først om det er en grunn til at produserbartDokument kommer som objekt

janitamarielle commented:
> Boksen med lagrede utkast som vises i `Send Brev` fanen skal vel også vises i `Dokumenter` fanen (ref skissene). Det gjør den ikke nå som jeg kan se

Bra sett! Har antatt den skissen bare nevnte endringene med ikonene i dokument-fanen (egen PR) 👍

janitamarielle commented:
> Får en error log når jeg trykker `Lagre og fortsett senere`: `SyntaxError: Unexpected end of JSON input`, men ser ikke at det faktisk skjer noe feil

Jeg klarer ikke reprodusere feilen. Gjør du noe spesifikt for å få denne frem?

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-4815_lagre-brev-som-utkast

 Conflicts:
	src/felleskomponenter/sideDialog/sendBrev/sendBrev.tsx

janitamarielle committed:
Oppdaterer oppsett tittelTilUtkast. Kan hende denne ikke blir nødvendig om den kommer fra backend

janitamarielle commented:
Setter til draft da jeg fortsatt må fullføre: 
- Bruke tittel fra backend
- Oversikt over utkast i dokument-fanen
- Fyll ut fritekstvedlegg basert på utkastet

janitamarielle committed:
Vise utkast fra dokument, og trigge visning av utkast ved trykk på lik linje som i send brev.
Flytter visning av lagrede utkast ut i egen komponent lagredeUtkast.tsx.
Flyttet da også aktivtUtkast ut i redux slik at både sideDialogDokumenter.tsx og sendBrev.tsx kan trigge valgt utkast.

janitamarielle committed:
Bruker tittel fra utkast-dtoen istedenfor å finne den selv.

janitamarielle committed:
Setter vedlegg fra utkast.

janitamarielle committed:
Unødvendig connect

janitamarielle committed:
Fjerner unødvendig boilerplate

martonSkjevelandNav reviewed (approved):
👏

janitamarielle committed:
Merge branch 'master' of https://github.com/navikt/melosys-web into MELOSYS-4815_lagre-brev-som-utkast

 Conflicts:
	src/services/api.ts
